apps/studio/src/server/modules/auth/email/__tests__/email.service.test.ts

```diff
@@ -0,0 +1,53 @@
+import { AuditLogEvent } from "@prisma/client"
+import { resetTables } from "tests/integration/helpers/db"
+import { setupUser } from "tests/integration/helpers/seed"
+
+import { db } from "~/server/modules/database"
+import { upsertUser } from "../email.service"
+
+describe("email.service", () => {
+  const TEST_EMAIL = "test@example.com"
+
+  beforeEach(async () => {
+    await resetTables("AuditLog", "User")
+  })
+
+  describe("upsertUser", () => {
+    it("should return an existing user if it already exists in the database", async () => {
+      // Arrange
+      await setupUser({ email: TEST_EMAIL })
+
+      // Act
+      const user = await db.transaction().execute(async (tx) => {
+        return upsertUser({ tx, email: TEST_EMAIL })
+      })
+
+      // Assert
+      const auditLogs = await db.selectFrom("AuditLog").selectAll().execute()
+      expect(user).toBeDefined()
+      expect(user.email).toBe(TEST_EMAIL)
+      expect(auditLogs).toHaveLength(0)
```

Comment:
nit - right now user-related test is sandwiched between audit logs setup and test. Perhaps re-order the code?

Reply:
Updated in 6df2f491137cb7321046a905680739bb66f3f381.